src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubScaleMonitor.cs

```diff
@@ -0,0 +1,395 @@
+            {
+                long partitionUnprocessedEventCount = 0;
+
+                Tuple<BlobPartitionLease, string> partitionLeaseFile = await GetPartitionLeaseFileAsync(i);
+                BlobPartitionLease partitionLeaseInfo = partitionLeaseFile.Item1;
+                string error = partitionLeaseFile.Item2;
+
+                if (partitionRuntimeInfo[i] == null || partitionLeaseInfo == null)
+                {
+                    partitionErrors.Add(error);
+                }
+                else
+                {
+                    // Check for the unprocessed messages when there are messages on the event hub parition
+                    // In that case, LastEnqueuedSequenceNumber will be >= 0
+                    if ((partitionRuntimeInfo[i].LastEnqueuedSequenceNumber != -1 && partitionRuntimeInfo[i].LastEnqueuedSequenceNumber != partitionLeaseInfo.SequenceNumber)
+                        || (partitionLeaseInfo.Offset == null && partitionRuntimeInfo[i].LastEnqueuedSequenceNumber >= 0))
+                    {
+                        partitionUnprocessedEventCount = GetUnprocessedEventCount(partitionRuntimeInfo[i], partitionLeaseInfo);
+                        totalUnprocessedEventCount += partitionUnprocessedEventCount;
+                    }
+                }
+
+                if (logPartitionInfo)
+                {
+                    _logger.LogInformation($"Function '{_functionId}', Partition: {i}, Total new partition events: {partitionUnprocessedEventCount}");
+                }
+            }
+
+            // Only log if not all partitions are failing or it's time to log
+            if (partitionErrors.Count > 0 && (partitionErrors.Count != partitionRuntimeInfo.Count || logPartitionWarning))
+            {
+                _logger.LogWarning($"Function '{_functionId}': Unable to deserialize partition or lease info with the " +
+                    $"following errors: {string.Join(" ", partitionErrors)}");
+                _nextPartitionWarningTime = DateTime.UtcNow.AddMinutes(PartitionLogIntervalInMinutes);
+            }
+
+            if (totalUnprocessedEventCount > 0 && logPartitionInfo)
+            {
+                _logger.LogInformation($"Function '{_functionId}', Total new events: {totalUnprocessedEventCount}");
+                _nextPartitionLogTime = DateTime.UtcNow.AddMinutes(PartitionLogIntervalInMinutes);
+            }
+
+            return new EventHubTriggerMetrics
+            {
+                Timestamp = DateTime.UtcNow,
+                PartitionCount = partitionRuntimeInfo.Count,
+                EventCount = totalUnprocessedEventCount
+            };
+        }
+
+        private async Task<Tuple<BlobPartitionLease, string>> GetPartitionLeaseFileAsync(int partitionId)
+        {
+            BlobPartitionLease blobPartitionLease = null;
+            string prefix = $"{EventHubOptions.GetBlobPrefix(_eventHubName, _connectionStringBuilder.Endpoint.Host)}{_consumerGroup}/{partitionId}";

```

Comment:
Does the prefix being used here need to match the prefix we passed into the EPH [here](https://github.com/Azure/azure-functions-eventhubs-extension/blob/371e9ff6b8b5eb24c5300e99b643b104c340533d/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Config/EventHubOptions.cs#L256)? If so, we need to be using the same shared logic to calculate it.

Reply:
Correct -- both this and the code you linked are using the same static `GetBlobPrefix` function to get the prefix. I also changed it to use another static function for getting the host string. Other than that, I couldn't find an instance where we're calculating the full storage path because I believe EPH does that internally